pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ConnectionFactory.java

```diff
@@ -28,7 +28,8 @@
  * Creates connections to Pinot, given various initialization methods.
  */
 public class ConnectionFactory {
-  private static PinotClientTransport _defaultTransport;
+  private static volatile PinotClientTransport _defaultTransport;
```

Comment:
(minor) This doesn't need to be volatile because it is already guarded by a lock

Reply:
Ack